gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -61,6 +62,9 @@ public CopyDataPublisher(State state) throws IOException {
     String uri = this.state.getProp(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, ConfigurationKeys.LOCAL_FS_URI);
 
     this.fs = FileSystem.get(URI.create(uri), conf);
+
+    FileAwareInputStreamDataWriterBuilder.setJobSpecificOutputPaths(state);
```

Comment:
The method `setJobSpecificOutputPaths` is setting the writer staging + output directories for the entire job correct? Why not just set this once in `Source` class, specifically in the given `SourceState` object?